src/faldbt/lib.py

```diff
@@ -88,14 +87,12 @@ def _execute_sql(
     sql: str,
     profile_target: str = None,
     config: RuntimeConfig = None,
+    adapter: Optional[SQLAdapter] = None,
 ) -> Tuple[AdapterResponse, RemoteRunResult]:
-    adapter = _get_adapter(project_dir, profiles_dir, profile_target, config)
-
-    # HACK: we need to include uniqueness (UUID4) to avoid clashes
-    name = "SQL:" + str(hash(sql)) + ":" + str(uuid4())
-    result = None
-    with adapter.connection_named(name):
-        response, execute_result = adapter.execute(sql, auto_begin=True, fetch=True)
+    def __exec(adapter: SQLAdapter, transaction: bool):
```

Comment:
>use a single leading underscore unless you have a really compelling reason to do otherwise (and even then think twice)

https://stackoverflow.com/a/41762806

Reply:
What does transaction do?

Reply:
> What does transaction do?

Adds `BEGIN` and `COMMIT` statements at the start and end of the SQL run.